iceoryx_hoofs/test/moduletests/test_cxx_string.cpp

```diff
@@ -2138,6 +2191,44 @@ TYPED_TEST(stringTyped_test, ConcatenateStringLiteralAndStringWithOperatorPlusWo
     EXPECT_THAT(testString.c_str(), StrEq("AdmTasse"));
 }
 
+TYPED_TEST(stringTyped_test, ConcatenateEmptyStringAndCharWithOperatorPlusWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "09efee5c-985f-4852-aa3c-5c67fc61071b");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    string<2U * STRINGCAP> testString = this->testSubject + 'M';
+    EXPECT_THAT(testString.capacity(), Eq(2U * STRINGCAP));
+    EXPECT_THAT(testString.size(), Eq(1U));
+    EXPECT_THAT(testString.c_str(), StrEq("M"));
+}
+
+TYPED_TEST(stringTyped_test, ConcatenateCharAndStringWithOperatorPlusWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "7d25ab82-48f9-4000-9764-e14522fd4ecc");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    this->testSubject = "S";
+    string<STRINGCAP + 7U> testString = 'F' + this->testSubject;
+    EXPECT_THAT(testString.capacity(), Eq(STRINGCAP + 7U));
+    EXPECT_THAT(testString.size(), Eq(2U));
+    EXPECT_THAT(testString.c_str(), StrEq("FS"));
+}
+
+TEST(String10, ConcatenateSeveralCharsAndStringsWithOperatorPlusWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "68e69585-fe83-497c-9249-59644501abd1");
+    const string<10U> testString1("Hyp");
+    const char testChar1 = 'n';
+    const string<10U> testString2("ot");
+    const char testChar2 = 'o';
+    const string<10U> testString3("ad");
+    auto result = testString1 + testChar1 + testString2 + testChar2 + testString3;
+
+    EXPECT_THAT(result.capacity(), Eq(32U));
```

Comment:
Here we need the capacity check (`auto`). As far as I can see, it adds all capacities to determine the capacity of the final string. I do not think we can do better with compile time knowledge only (and we need the type at this time). 

Maybe we can vary the capacity of one of the strings to be something other than 10? I am fine with the magic number 32 (test is short enough) but it would change of course if one of the strings has capacity != 10.